tests/test_config.py

```diff
@@ -0,0 +1,87 @@
+import abc
+
+import numpy as np
+import pytest
+
+import probnum
+from probnum import randvars
+
+
+class _TestConfig(abc.ABC):
+    @abc.abstractmethod
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        pass
+
+    def test_contextmanager(self):
+        none_vals = {key: None for (key, _) in self.defaults}
+
+        for key, default_val in self.defaults:
+            # Check if default is correct before context manager
+            assert getattr(probnum.config, key) == default_val
+            # Temporarily set all config values to None
+            with probnum.config(**none_vals):
+                assert getattr(probnum.config, key) is None
+
+            # Check if the original (default) values are set after exiting the context
+            # manager
+            assert getattr(probnum.config, key) == default_val
+
+
+class TestRandvarConfig(_TestConfig):
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        self.defaults = [
+            ("covariance_inversion_damping", 1e-12),
+        ]
+
+    @pytest.fixture
+    def zero_cov_normal(self):
+        return randvars.Normal(np.random.rand(5), np.zeros((5, 5)))
+
+    def test_randvars_config(self, zero_cov_normal):
+        chol_1 = zero_cov_normal.dense_cov_cholesky()
```

Comment:
maybe via `precompute_cov_cholesky()` here? I am not sure how public the `dense_cov_cholesky()` was intended to be (but I didnt write it, so I am not the expert here).

Reply:
`dense_cov_cholesky` is meant to be this public :) It helps to reduce code if you're not shure whether the covariance is an `ndarray` or a `LinearOperator`

Reply:
ayt